docs/tutorials/rest_api/index.md

```diff
@@ -0,0 +1,405 @@
+
+### Step 3: Pipeline creation
+
+We create the pipeline referring by name to the program and all
+the connectors we have created previously. Each connector is
+attached as input to their respective stream table.
+
+```
+curl -i -X PUT http://localhost:8080/v0/pipelines/sc-pipeline \
+-H 'Content-Type: application/json' \
+-d '{
+    "description": "Supply Chain pipeline",
+    "program_name": "sc-program",
+    "config": {"workers": 4},
+    "connectors": [
+         {
+             "connector_name": "sc-connector-part",
+             "is_input": true,
+             "name": "sc-connector-part",
+             "relation_name": "PART"
+         },
+         {
+             "connector_name": "sc-connector-vendor",
+             "is_input": true,
+             "name": "sc-connector-vendor",
+             "relation_name": "VENDOR"
+         },
+         {
+             "connector_name": "sc-connector-price",
+             "is_input": true,
+             "name": "sc-connector-price",
+             "relation_name": "PRICE"
+         }
+    ]
+}'
+```
+
+> Observe in the above format that it is possible to connect multiple
+> connectors to the same stream table or incremental view.
+
+It will return `HTTP/1.1 201 Created` when successful.
+The pipeline will now be ready to be started.
+
+### Step 4: Starting pipeline
+
+We start the pipeline using:
+
+```
+curl -i -X POST http://localhost:8080/v0/pipelines/sc-pipeline/start
+```
+
+... which will return `HTTP/1.1 202 Accepted` when successful.
+
+Check it is successfully started using:
+
+```
+curl -X GET http://localhost:8080/v0/pipelines/sc-pipeline
+```
+
+... which will have `current_status` set to `Running` when it has started.
+
+> Note: only when a pipeline is started, the connectors are actually used.
+> A pipeline will not start if a connector is unable to connect to its
+> data source or sink (e.g., if a URL is misspelled).
+
+> To restart a pipeline (e.g., to have updates to its program or schema
+> take effect):
+> ```
+> # Shut it down:
+> curl -i -X POST http://localhost:8080/v0/pipelines/sc-pipeline/shutdown
+> # ... wait for the current_status to become Shutdown by checking:
+> curl -X GET http://localhost:8080/v0/pipelines/sc-pipeline
+> # ... and then start:
+> curl -i -X POST http://localhost:8080/v0/pipelines/sc-pipeline/start
+> ```
+
+### Step 5: Pipeline progress
+
+A running pipeline provides a multitude of interesting stats:
+
+```
+curl -X GET http://localhost:8080/v0/pipelines/sc-pipeline/stats
+```
+
+... such as the number of input and processed records.
+The `total_processed_records` should be `9` (3 rows each for part, vendor, and price).
+
+### Step 6: Read data directly from an incremental view
+
+Both input and output connectors are optional, in the sense that input and
+output of data can directly be done using HTTP requests as well.
+
+We can retrieve the incremental view quantiles snapshot using `curl`:
```

Comment:
Quantiles might be too much of an advanced topic for this tutorial.  They would require some more explanation.  Or I think it's ok to just skip them.

Reply:
I'm in favor of keeping it just because the ability to view a view via curl is I think important functionality to showcase. When the full snapshot retrieval is possible, we should update it.